src/snark_wrapper_verifier.rs

```diff
@@ -95,3 +111,90 @@ pub async fn generate_solidity_test(args: &GenerateSolidityTestArgs) -> Result<(
     println!("recursiveAggregationInput: [] \n }};");
     Ok(())
 }
+
+pub async fn verify_snark_from_l1(
```

Comment:
nit: surprised that verify_snark and verify_snark_from_l1 don't share more code.